blockchain/chain.go

```diff
@@ -1309,6 +1412,7 @@ func New(config *Config) (*BlockChain, error) {
 		blocksPerRetarget:   int32(targetTimespan / targetTimePerBlock),
 		index:               newBlockIndex(config.DB, params),
 		hashCache:           config.HashCache,
+		bestChain:           newChainView(nil),
```

Comment:
I'm fairly certain that is what happens already.  Please double check me, but this will just result in a chain view with a nil backing array (no allocations).  Then, when the initialization happens, it calls `.SetTip` which will notice that the cap of the nil slice is less than the needed amount and perform the single allocation then (along with and additional `approxNodesPerWeek`) for future expansion before a realloc is needed.